### ISSUE TITLE
Hang in SolutionBasedGraphCanMatchProjectSpecificConfigurations test

### ISSUE THREAD

**Issue by rainersigwald** [microsoft/msbuild#5453]
Observed in https://dev.azure.com/dnceng/public/_build/results?buildId=698758

    Microsoft.Build.Engine.UnitTests: [Long Running Test] 'Microsoft.Build.Graph.UnitTests.GraphLoadedFromSolutionTests.SolutionBasedGraphCanMatchProjectSpecificConfigurations', Elapsed: 00:46:11

**Comment by rainersigwald** [comment]
@cdmihai anything spring to mind about this?

**Comment by cdmihai** [comment]
Funky. Looking over what the test does:
- it constructs a graph
- uses that graph and some other test inputs to generate a solution file
- it constructs a graph based on the generated solution file and asserts some properties

The hang risky places in all of this is the parallel graph construction, given the amount of synchronization in ParallelWorkSet.

But I have no concrete leads. It would be awesome if there was a way to capture a process dump on a hanged test, so we can look at the callstacks

**Comment by Forgind** [comment]
Have you made any progress on this? Hit it multiple times today. 😢

**Comment by cdmihai** [comment]
Nope, tried to trigger it a few times in its debug PR but didn't repro it. Don't have much time right now so I'll just disable the test and link it here.
